pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -154,6 +154,8 @@ func (r *DNSPolicyReconciler) reconcileResources(ctx context.Context, dnsPolicy
 		return err
 	}
 
+	dnsPolicy.Default()
```

Comment:
Is this a pattern used elsewhere? seems odd to me to change the values of something that is already created. Is there some reason we would do this over setting a default value in the CRD? see `defaultWeight` [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/apis/v1alpha1/dnspolicy_types.go#L66) and [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/config/crd/bases/kuadrant.io_dnspolicies.yaml#L148)

Reply:
It is a pattern used by the webhooks. Basically by implementing the `Default` and `Validate` method, it's very straightforward to setup webhooks for the CR. I thought I'd implement them now so we can reuse them if we want to create webhooks for the DNSPolicy, but I agree we should use the default value in the CRD as well

Reply:
Presumably that would normally be called in a mutating webhook before the object is ever stored in etcd. Just seems odd to call it as part of the reconciliation loop.

Reply:
Yes, I'd remove that call from the reconciliation loop when the webhook is added, so it's not duplicated, but as we don't have the webhook yet I put it in the reconcile, so that any following logic can assume that all values have been set